clear_costmap_recovery/src/clear_costmap_recovery.cpp

```diff
@@ -162,6 +167,12 @@ void ClearCostmapRecovery::clear(costmap_2d::Costmap2DROS* costmap){
       clearMap(costmap, x, y);
     }
   }
+
+  for (const auto& layer : clearable_layers_) {
+    if (!cleared_layers.count(layer)) {
+      ROS_WARN_STREAM("Cannot clear layer " << layer << " because it does not exist");
```

Comment:
S: technically, there is `ROS_WARN_STREAM_COND` for exactly this use case (but it is somewhat debatable whether it improves readability or not)

Reply:
Yeah I like your suggestion better!